src/button-group/interfaces.ts

```diff
@@ -0,0 +1,114 @@
+   *
+   * However, use discretion. We recommend you don't enable this property unless necessary
+   * because fixed positioning results in a slight, visible lag when scrolling complex pages.
+   */
+  dropdownExpandToViewport?: boolean;
+  /**
+   * Array of objects of type "icon-button", "group", or "menu-dropdown".
+   *
+   * ### type="icon-button"
+   * * `id` (string) - The unique identifier of the button, used as detail in `onItemClick` handler and to focus the button using `ref.focus(id)`.
+   * * `text` (string) - The name shown as a tooltip or menu text for this button.
+   * * `loading` (optional, boolean) - The loading state indication for the button.
+   * * `loadingText` (optional, string) - The loading text announced to screen-readers.
+   * * `disabled` (optional, boolean) - The disabled state indication for the button.
+   * * `iconName` (optional, string) - Specifies the name of the icon, used with the [icon component](/components/icon/).
+   * * `iconAlt` (optional, string) - Specifies alternate text for the icon when using `iconUrl`.
+   * * `iconUrl` (optional, string) - Specifies the URL of a custom icon.
+   * * `iconSvg` (optional, ReactNode) - Custom SVG icon. Equivalent to the `svg` slot of the [icon component](/components/icon/).
+   * * `feedbackText` (optional, string) - Text that appears when the user clicks the button. Use to provide feedback to the user.
+   *
+   * ### type="group"
+   * * `text` (string) - The name of the group rendered as ARIA label for this group.
+   * * `items` ((ButtonGroupProps.IconButton | ButtonGroupProps.MenuDropdown)[]) - The array of items that belong to this group.
+   *
+   * ### type="button-dropdown"
+   * * `id` (string) - The unique identifier of the button, used to focus the button using `ref.focus(id)`.
+   * * `text` (string) - The name of the menu button shown as a tooltip.
+   * * `loading` (optional, boolean) - The loading state indication for the menu button.
+   * * `loadingText` (optional, string) - The loading text announced to screen-readers.
+   * * `disabled` (optional, boolean) - The disabled state indication for the menu button.
+   * * `items` (ButtonDropdownProps.ItemOrGroup[]) - The array of dropdown items that belong to this menu.
+   */
+  items: ReadonlyArray<ButtonGroupProps.ItemOrGroup>;
+  /**
+   * Called when the user clicks on an item, and the item is not disabled. The event detail object contains the id of the clicked item.
+   */
+  onItemClick?: NonCancelableEventHandler<ButtonGroupProps.ItemClickDetails>;
+}
+
+export interface InternalButtonGroupProps extends ButtonGroupProps, InternalBaseComponentProps {}
+
+export namespace ButtonGroupProps {
+  // Fixes documenter output.
+  // When a union only has one element the resulting type for it is not string but object.
+  export type Variant = 'icon' | 'icon';
+
+  export type ItemOrGroup = Item | Group;
+  export type Item = IconButton | MenuDropdown;
```

Comment:
Let's make the types Item and ItemOrGroup private (move outside the namespace)

Reply:
Why do you want to do it? It ssems that users should be able to create items of these types

Reply:
Ah, I forgot the type is used in the interface above. Nevermind - let's keep both Item and ItemOrGroup